code/datums/tutorial/marine/hospital_corpsman_sandbox.dm

```diff
@@ -0,0 +1,682 @@
+		dragging_agents -= dragging_agent
+	dragging_agent.density = 0
+	QDEL_IN(dragging_agent, 2.5 SECONDS)
+	animate(dragging_agent, 2.5 SECONDS, alpha = 0, easing = CUBIC_EASING)
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/proc/move_active_agents()
+
+	listclearnulls(active_agents) // failsafe
+	for(var/mob/living/carbon/human/realistic_dummy/active_agent as anything in active_agents)
+		var/turf/dropoff_point = agents[active_agent]
+		if(locate(active_agent) in agent_spawn_location)
+			var/initial_step_direction = pick((agent_spawn_location.y) <= (dropoff_point.y) ? NORTH : SOUTH)
+			var/turf/target_turf = get_step(active_agent, initial_step_direction)
+			active_agent.Move(target_turf, initial_step_direction)
+		if((dropoff_point.x > active_agent.x) || (dropoff_point.x < active_agent.x))
+			var/step_direction = pick((active_agent.x) > (dropoff_point.x) ? WEST : EAST)
+			var/turf/target_turf = get_step(active_agent, step_direction)
+			active_agent.Move(target_turf, step_direction)
+		else if(dropoff_point.x == active_agent.x)
+			if(((dropoff_point.y - active_agent.y) >= 2) || ((dropoff_point.y - active_agent.y) <= (-2)))
+				var/step_direction = pick((active_agent.y) > (dropoff_point.y) ? SOUTH : NORTH)
+				var/turf/target_turf = get_step(active_agent, step_direction)
+				active_agent.Move(target_turf, step_direction)
+				return
+			active_agent.Move(dropoff_point)
+			active_agents -= active_agent
+			handle_speech(active_agent)
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/proc/make_agent_leave(mob/living/carbon/human/realistic_dummy/agent, bypass)
+	SIGNAL_HANDLER
+
+	UnregisterSignal(agent, COMSIG_LIVING_REJUVENATED)
+	UnregisterSignal(agent, COMSIG_HUMAN_SET_UNDEFIBBABLE)
+	UnregisterSignal(agent, COMSIG_HUMAN_TUTORIAL_HEALED)
+	agent.updatehealth()
+	if(!(bypass))
+		if(agent.undefibbable == TRUE)
+			agent.balloon_alert_to_viewers("[agent.name] permanently dead!", null, DEFAULT_MESSAGE_RANGE, null, COLOR_RED)
+			playsound(agent.loc, 'sound/items/defib_failed.ogg', 20)
+		else
+			agent.balloon_alert_to_viewers("[agent.name] fully treated!")
+			playsound(agent.loc, 'sound/machines/terminal_success.ogg', 20)
+	agents -= agent
+	if(agent in active_agents) // failsafe in case patient NPC was healed, despite never reaching their dropzone
+		active_agents -= agent
+	QDEL_IN(agent, 2.5 SECONDS)
+	animate(agent, 2.5 SECONDS, alpha = 0, easing = CUBIC_EASING)
+	for(var/obj/item/clothing/suit/storage/marine/medium/armor in cleanup)
+		item_cleanup(armor)
+	if((length(agents)) == 0)
+		INVOKE_ASYNC(src, PROC_REF(handle_round_progression))
+
+/datum/tutorial/marine/hospital_corpsman_sandbox/proc/eval_booboo_agent()
+	SIGNAL_HANDLER
+
+	var/mob/living/carbon/human/realistic_dummy/active_agent = new(agent_spawn_location)
+	arm_equipment(active_agent, /datum/equipment_preset/uscm/tutorial_rifleman)
+	var/turf/dropoff_point = loc_from_corner(round(rand(6, 8), 1), round(rand(1, 3)))
```

Comment:
when given a lower and upper bound rand() returns an integer so round() is not needed https://www.byond.com/docs/ref/#/proc/rand